docs/operations-guide/upgrading-metabase.md

```diff
@@ -1,31 +1,68 @@
 
-If you are running Metabase via docker, then you simply need to stop the Docker container and start a new container with the latest Metabase image. On startup, Metabase will perform any upgrade tasks it needs to perform, and once it's finished you'll be running the new version.
+Steps differ depending on whether you're running the JAR or a Docker image.
 
-To pull the latest Metabase:
+- [Upgrading the JAR](#upgrading-the-jar)
+- [Upgrading the Docker image](#upgrading-the-docker-image)
 
-    $ docker pull metabase/metabase:latest
+### Upgrading the JAR
 
-### Jar file
+If you're running the JVM Jar file directly:
 
-If you are running the JVM Jar file directly, then you simply kill the process, replace the .jar file with the newer version and restart the server. On startup, Metabase will perform any upgrade tasks it needs to perform, and once it's finished you'll be running the new version.
+1. [Back up your application database](backing-up-metabase-application-data.md).
 
-### [Upgrading AWS Elastic Beanstalk deployments](running-metabase-on-elastic-beanstalk.html#deploying-new-versions-of-metabase-on-elastic-beanstalk)
+2. [Download the latest version](https://www.metabase.com/start/oss/jar.html).
 
-Step-by-step instructions on how to upgrade Metabase running on Elastic Beanstalk using RDS.
+3. Use a terminal to access your existing Metabase process and kill it (usually CTRL-C).
 
-### [Upgrading Azure Web Apps deployments](running-metabase-on-azure.html#additional-configurations)
+4. Replace the existing JAR file (`metabase.jar`) in your Metabase directory with the newer version.
 
-Step-by-step instructions on how to upgrade Metabase running on Azure Web Apps.
+5. Restart the server:
 
-### [Upgrading Heroku deployments](running-metabase-on-heroku.html#deploying-new-versions-of-metabase)
+```
+java -jar metabase.jar
+```
 
-Step-by-step instructions on how to upgrade Metabase running on Heroku.
+On startup, Metabase will perform any tasks it needs to complete the upgrade. Once Metabase has completed those tasks, you'll be running the new version.
+
+### Upgrading the Docker image
+
+If you're running Metabase in a Docker container:
+
+1. [Back up your application database](backing-up-metabase-application-data.md). 
+
+> WARNING: If you're not using a [production-ready database](migrating-from-h2.md), your application data (questions, dashboards, and so on) will have been stored in an H2 database _inside_ your container. Upgrading requires swapping out your existing container for a new image with the upgraded Metabase JAR, which will wipe out your application data. We recommend switching to a production-ready database before you upgrade. 
+
+2. Stop the current Docker container.
+
+3. Pull the latest Metabase Docker image:
+
+```
+docker pull metabase/metabase:latest
+```
+
+4. Start the new Docker container.  Depending on the ports and what you want to name the container, the command will look something like: Something like:
```

Comment:
Should probably link to /docs/operations-guide/running-metabase-on-docker.md#production-installation, since it just needs to be the same setup as what was originally made, when doing the production setup.